### ISSUE TITLE
Improve sshd configuration

### ISSUE THREAD

**Issue by willthames** [bottlerocket-os/bottlerocket-admin-container#41]
**Image I'm using:**

Custom image based on pr #39 but that should be pretty much latest now.

**Issue or Feature Request:**

I would like either a stricter sshd config or the ability to configure a stricter sshd config.

e.g. 

```
Ciphers chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
PermitRootLogin no
```

I can see two ways to achieve that:
* hardcode it for everyone in sshd_config
* add a `custom-config` (or better name) to user data e.g.
  ```
  {
    "ssh":{
       "custom-config": {
          "PermitRootLogin": "no",
          "Ciphers": "chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com"
       }
    }
  }
  ```

I am willing to implement this but just want to check what your preferences are first.

**Comment by jpculp** [comment]
@willthames, thank you for opening this enhancement request! After some discussion with the team, we agree that this would be a great change. We would prefer hardcoding the `PermitRootLogin no` directly to the `sshd_config` above the `# Configured by user data` line, and setting the ciphers via user-data like so:
```
{
    "ssh": {
        "authorized-keys-command": "/opt/aws/bin/eic_run_authorized_keys %u %f",
        "authorized-keys-command-user": "ec2-instance-connect",
        "ciphers": [
            "chacha20-poly1305@openssh.com",
            "aes128-ctr",
            "aes192-ctr",
            "aes256-ctr",
            "aes128-gcm@openssh.com",
            "aes256-gcm@openssh.com"
        ]
    }
}
```
Is this something you would still be willing to implement or would you rather us work on it?

**Comment by willthames** [comment]
@jpculp - I'm happy with that approach - I'll hopefully submit a PR shortly

**Comment by willthames** [comment]
Closed by #42
